internal/pkg/source/source.go

```diff
@@ -232,25 +233,49 @@ func sourcesFromBlocks(fn *ssa.Function, conf classifier) []*Source {
 
 		for _, instr := range b.Instrs {
 			switch v := instr.(type) {
-			// Looking for sources of PII allocated within the body of a function.
+			default:
+				continue
+
+			// source variable allocated
 			case *ssa.Alloc:
-				if conf.IsSource(utils.Dereference(v.Type())) && !isProducedBySanitizer(v, conf) {
-					sources = append(sources, New(v, conf))
+				if isProducedBySanitizer(v, conf) {
+					continue
 				}
 
-				// Handling the case where PII may be in a receiver
-				// (ex. func(b *something) { log.Info(something.PII) }
+			// source returned from call
+			case *ssa.Call:
+				if isProducedBySanitizer(v, conf) {
+					continue
+				}
+
+			// source obtained from field
 			case *ssa.FieldAddr:
-				if conf.IsSource(utils.Dereference(v.Type())) {
-					sources = append(sources, New(v, conf))
+
+			// source obtained from array by index
+			case *ssa.IndexAddr:
+
+			// source obtained from map by key
+			case *ssa.Lookup:
+
```

Comment:
Are these intentionally no-ops?

Reply:
Yep. The intent is that in all other cases the `default` will get hit, which will cause the `for` to loop. Hitting one of the `case`s will lead to the code after the switch being executed. I did this because every case is pretty much the same otherwise (lots of duplicated code).

From your comment though I understand that this way of structuring the code way be misleading. Would you prefer that I duplicate the code? Or that I find some other way to avoid the duplication?

Reply:
Ha!  I guess I'm acclimated to seeing the `default` at the bottom, I just banner-blindness'd right past it.  (It's fine where it is, just my own patterns.)

Stylistically, Go does allow multiple cases in-line.  You could consider collapsing the no-op cases

```golang
			switch v := instr.(type) {
			// Personally, I could go either way on collecting these by case, since it requires a type-assertion.  Just demonstrating here.
			case *ssa.Alloc, *ssa.Call:
				if isProducedBySanitizer(v.(ssa.Value), conf) {
					continue
				}

			// source obtained from field
			case *ssa.FieldAddr, *ssa.IndexAddr, *ssa.Lookup:
				// pass
```

The intent of the switch is to act as a filter, then?  "Get Allocs and Calls, unless they're from a sanitizer.  Get unary operator values other than channels.  Get these three.  Skip everything else."

Are we intentionally skipping `Value`s that don't fit the above?

All this to say, a docstring for reference on the function would be helpful now that it's grown in complexity.

Reply:
For some reason I can reply to your comment inline now.

>The intent of the switch is to act as a filter, then? "Get Allocs and Calls, unless they're from a sanitizer. Get unary operator values other than channels. Get these three. Skip everything else."
>Are we intentionally skipping Values that don't fit the above?

Yep! We may want to add more cases eventually, but for now these cover our tests and I can't think of other compelling cases. There might be the Field instruction, but I've never seen it in our ssa dumps (I do see plenty of FieldAddrs though). There might also be the Extract instruction, but in the tests I ran it was paired with other instructions that we already handle, so I don't think we need it.

>Stylistically, Go does allow multiple cases in-line. You could consider collapsing the no-op cases

Ah, very nice. I tried to do that one time, and the code wouldn't compile, so I thought type switches were somehow different from regular switches and you couldn't have many types in one case. Turns out, the error I got that time was due to the fact that when you use multiple types in a case, the type you get is the original type of the value you're switching on, so for Alloc and Call, the type was ssa.Instruction, and I was using it as an ssa.Value in the body of the case (they are both ssa.Values, but the compiler doesn't check for that). Anyhoo. Thanks, reads much better now!

>All this to say, a docstring for reference on the function would be helpful now that it's grown in complexity.

Good point. Done.